k8s/core/test5/betterstack/pkg/betterstatus/betterstatus.go

```diff
@@ -16,33 +18,68 @@ type BetterStackApiCaller struct {
 
 func (bsapi *BetterStackApiCaller) marshalJson(payload interface{}) (jsonBody []byte, err error) {
 	switch concretePayload := payload.(type) {
-	case GnoMonitorPayload, CreateMonitorGroupPayload, CreateMonitorPayload:
+	case GnoMonitorPayload, CreateMonitorGroupPayload, CreateMonitorPayload, CreateStatusPageSectionPayload, CreateStatusPageResourcePayload:
 		jsonBody, err = json.Marshal(concretePayload)
 	default:
 		jsonBody, err = json.Marshal(payload)
 	}
 	return
 }
 
-func (bsapo *BetterStackApiCaller) handleHttpResp(endpoint string, resp *http.Response) (respBody []byte, err error) {
+func (bsapi *BetterStackApiCaller) unmarshalJson(respBody []byte, respReceviver interface{}) error {
+	switch respReceviver.(type) {
+	case *CreateMonitorGroupResponse, *CreateMonitorResponse, *CreateStatusPageSectionResponse:
+		decoder := json.NewDecoder(bytes.NewReader(respBody))
+		if err := decoder.Decode(&respReceviver); err != nil {
+			return fmt.Errorf("Unable to parse body: %w", err)
+		}
+		return nil
```

Comment:
```suggestion
		if err := json.Unmarshal(respBody, &respReceiver); err != nil {
    			return fmt.Errorf("Unable to parse body: %w", err)
		}
		
		return nil
```

Or something similar, but didn't compile the code.

Reply:
let's see if the updated code makes more sense